### ISSUE TITLE
Redirect smart card only "armhf"

### ISSUE THREAD

**Issue by mityajko94** [FreeRDP/FreeRDP#5397]
Hello guys. Early sorry, i bad speak english.
Have truble - redirect smartcard only Asus TinkerBoard.
Linux tinkerboard 4.4.132+ #1 SMP Tue Oct 23 18:03:49 CST 2018 armv7l GNU/Linux

We retry with /smartcard:all -  start smartcard redirecting - yes, then it break.  It not read tokken.
But , on amd64 archi - all ok. 
Cryptoprovider  - JaCarta, eTokken 
```
This is FreeRDP version 2.0.0-dev5 (8dcac1ada)
Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=ON HAVE_AIO_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_INTTYPES_H=1 HAVE_JOURNALD_H=TRUE HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL= HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_XI_TOUCH_CLASS=1 WITH_ALSA=ON WITH_CAIRO=OFF WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=on WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=OFF WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=TRUE WITH_FFMPEG=TRUE WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=OFF WITH_GSSAPI=OFF WITH_GSTREAMER_0_10=OFF WITH_GSTREAMER_1_0=ON WITH_ICU=OFF WITH_JPEG=OFF WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_MACAUDIO=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_NEON=ON WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PCSC=OFF WITH_PROFILER=OFF WITH_PULSE=on WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=OFF WITH_SERVER_INTERFACE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SOXR=OFF WITH_SSE2=OFF WITH_SWSCALE=OFF WITH_THIRD_PARTY=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=ON WITH_X11=ON WITH_X264=OFF WITH_XCURSOR=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XSHM=ON WITH_XV=ON WITH_ZLIB=ON
Build type:          Debug
CFLAGS:               -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -g -DWINPR_DLL
Compiler:            GNU, 6.3.0
Target architecture: ARM
```

**Comment by akallabeth** [comment]
Connection log (with /log-level:debug) and commandline might help to diagnose.
Just a question: same distro on both or different ones? (A test with the same software, that means dependencies of freerdp, on amd64 and your arm system might eliminate external issues)

**Comment by mityajko94** [comment]
> Connection log (with /log-level:debug) and commandline might help to diagnose.
> Just a question: same distro on both or different ones? (A test with the same software, that means dependencies of freerdp, on amd64 and your arm system might eliminate external issues)
on amd64 i test on virtualbox. it's all okay

on arm pcscd :1.8.20-1.   
run with xfreerdp /v:adress /u:test /p:12345 /smartcard /log-level:debug
`[06:39:24:641] [3756:3782] [DEBUG][com.freerdp.channels.smartcard.client] - BeginTransaction_Return {
[06:39:24:641] [3756:3782] [DEBUG][com.freerdp.channels.smartcard.client] - ReturnCode: SCARD_S_SUCCESS (0x00000000)
[06:39:24:641] [3756:3782] [DEBUG][com.freerdp.channels.smartcard.client] - }
[06:39:24:656] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - SCardBeginTransaction (0x000900BC) FileId: 1 CompletionId: 4
[06:39:24:656] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - BeginTransaction_Call {
[06:39:24:656] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - hContext: 0x30F7905A (4)
[06:39:24:656] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - hCard: 0x0CB8567F (4)
[06:39:24:656] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - dwDisposition: SCARD_LEAVE_CARD (0x00000000)
[06:39:24:656] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - }
[06:39:24:671] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - SCardStatusW (0x000900CC) FileId: 1 CompletionId: 1
[06:39:24:671] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - StatusW_Call {
[06:39:24:671] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - hContext: 0x0971832E (4)
[06:39:24:671] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - hCard: 0x9B307B31 (4)
[06:39:24:672] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - fmszReaderNamesIsNULL: 1 cchReaderLen: 0 cbAtrLen: 0
[06:39:24:672] [3756:3765] [DEBUG][com.freerdp.channels.smartcard.client] - }
[06:39:24:673] [3756:3782] [DEBUG][com.freerdp.utils] - fatal_handler: signum=11
Segmentation fault`

**Comment by akallabeth** [comment]
Fixed with #5830
